gprofiler/utils/node.py

```diff
@@ -0,0 +1,176 @@
+        )
+
+    response_json = debugger_url_response.json()
+    if (
+        not isinstance(response_json, list)
+        or len(response_json) == 0
+        or not isinstance(response_json[0], dict)
+        or "webSocketDebuggerUrl" not in response_json[0]
+    ):
+        raise NodeDebuggerUrlNotFound(response_json)
+
+    return response_json[0]["webSocketDebuggerUrl"]  # type: ignore
+
+
+@retry(NodeDebuggerUnexpectedResponse, 5, 1)
+def _send_socket_request(sock: WebSocket, cdp_request: Dict) -> None:
+    sock.settimeout(2)
+    sock.send(json.dumps(cdp_request))
+    message = sock.recv()
+    try:
+        message = json.loads(message)
+    except json.JSONDecodeError:
+        raise NodeDebuggerUnexpectedResponse(message)
+
+    if (
+        "result" not in message.keys()
+        or "result" not in message["result"].keys()
+        or "type" not in message["result"]["result"].keys()
+        or message["result"]["result"]["type"] != "boolean"
+    ):
+        raise NodeDebuggerUnexpectedResponse(message)
+
+
+def _load_dso(sock: WebSocket, module_path: str) -> None:
+    cdp_request = {
+        "id": 1,
+        "method": "Runtime.evaluate",
+        "params": {
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").start()',
+            "replMode": True,
+        },
+    }
+    _send_socket_request(sock, cdp_request)
+
+
+def _stop_dso(sock: WebSocket, module_path: str) -> None:
+    cdp_request = {
+        "id": 1,
+        "method": "Runtime.evaluate",
+        "params": {
+            "expression": f'process.mainModule.require("{os.path.join(module_path, "linux-perf.js")}").stop()',
+            "replMode": True,
+        },
+    }
+    _send_socket_request(sock, cdp_request)
+
+
+def _copy_module_into_process_ns(process: psutil.Process, musl: bool, version: str) -> str:
+    proc_root = get_proc_root_path(process)
+    libc = "musl" if musl else "glibc"
+    dest_inside_container = get_dest_inside_container(musl, version)
+    dest = resolve_proc_root_links(proc_root, dest_inside_container)
+    if os.path.exists(dest):
+        return dest_inside_container
+    src = resource_path(os.path.join("node", "module", libc, version))
+    shutil.copytree(src, dest)
+    add_permission_dir(dest, stat.S_IROTH, stat.S_IXOTH | stat.S_IROTH)
+    return dest_inside_container
+
+
+def _generate_perf_map(module_path: str) -> None:
+    debugger_url = _get_debugger_url()
+    sock = create_connection(debugger_url)
+    sock.settimeout(2)
+    _load_dso(sock, module_path)
+
+
+def _clean_up(module_path: str, pid: int) -> None:
+    debugger_url = _get_debugger_url()
+    sock = create_connection(debugger_url)
+    _stop_dso(sock, module_path)
+    sock.settimeout(2)
+    os.remove(os.path.join("/tmp", f"perf-{pid}.map"))
+
+
+def get_node_processes() -> List[psutil.Process]:
+    return pgrep_exe(r"(?:^.+/node[^/]*$)")
```

Comment:
```suggestion
    return pgrep_exe(r"(?:^.*/node[^/]*$)")
```

(to catch `/node` as well)

Also - what are the `(?: ... )` for (parenthesis)